### ISSUE TITLE
Client gets out of sync with server

### ISSUE THREAD

**Issue by krwminer** [SlimeKnights/TinkersConstruct#1527]
When a Forge Player Interact Event, (Right Click Block), is rejected when accessing the Smeltery on the server side of this mod, the client side gets out of sync and sometimes crashes depending on what you do inside the Smeltery GUI.

Same is true for a lot of other TiC container blocks like the casting tables, bins and faucets.

This was discovered when using MyTown2 protections which is a server side only mod.

**Comment by bonii-xx** [comment]
How do I best put this...
The right-click interaction event is Clientside only.

**Comment by krwminer** [comment]
It is not Clientside only. MyTown2 is server side mod only. We catch it server side and use it all the time for protection for many mods. Look it up at net.minecraftforge.event.entity.player.

If you want further info on it:

http://jabelarminecraft.blogspot.com/p/minecraft-forge-172-event-handling.html

Scroll down and look at PlayerInteractEvent.

Even this is a bit out of date, since we are able to get the RIGHT_CLICK_AIR events on the server side also.

**Comment by bonii-xx** [comment]
Sadly my code reading skills are a better source of information than some random wepage, and I can tell you that RIGHT_CLICK_BLOCK is not called for interacting with blocks serverside.

**Comment by AfterWind** [comment]
https://github.com/MyEssentials/MyTown2/blob/master/src/main/java/mytown/protection/Protections.java#L321

Proof that it works server-side on a server-side mod. And no, it's not only on placing the block.

**Comment by bonii-xx** [comment]
That's the source from MyTown2. That is no proof at all, show me where the event is fired, not where it's used.

**Comment by AfterWind** [comment]
Does it really matter where it's fired? Since it's getting handled even after verifying that it's on the server-side. Why does it matter where it fires? You can try the code for yourself. 

I concluded that you really don't want to listen to my suggestion even when I present you actual code that works even though you say it shouldn't. Sorry that I even bothered.

**Comment by krwminer** [comment]
![serverrightclickblock](https://cloud.githubusercontent.com/assets/4827567/7101817/a72ee6c6-e038-11e4-8c4a-769e1301cbca.jpg)

**Comment by bonii-xx** [comment]
My apologies, I found the reason why things didn't line up: there was a bug in the test setup that caused the event to not fire correctly. I was able to reproduce it now. I shouldn't debug when I'm tired..

It seems whoever wrote the stuff executed the actions both server and clientside, although they should only happen serverside. Above commit fixes this for the smeltery, faucet and casting table/basin. Just report back here if any more blocks have incorrect behaviour.
